internal/convenience/synchronizer/graphile_synchornizer.go

```diff
@@ -64,41 +71,29 @@ func (x GraphileSynchronizer) Start(ctx context.Context, ready chan<- struct{})
 
 }
 
-func (x GraphileSynchronizer) handleGraphileResponse(outputResp OutputResponse, ctx context.Context) {
+func (x GraphileSynchronizer) handleGraphileResponse(ctx context.Context, outputResp OutputResponse) error {
 	// Handle response data
-	voucherIds := []string{}
 	var initCursorAfter string
 	var initInputCursorAfter string
 	var initReportCursorAfter string
 
 	for _, output := range outputResp.Data.Outputs.Edges {
-		outputIndex := output.Node.Index
-		inputIndex := output.Node.InputIndex
-		slog.Debug("Add Voucher/Notices",
-			"inputIndex", inputIndex,
-			"outputIndex", outputIndex,
-		)
-		voucherIds = append(
-			voucherIds,
-			fmt.Sprintf("%d:%d", inputIndex, outputIndex),
-		)
-		adapted := adapter.ConvertVoucherPayloadToV2(
-			output.Node.Blob[2:],
-		)
-		destination, _ := adapter.GetDestination(output.Node.Blob)
+		processOutPutData, err := x.processOutput(output)
```

Comment:
I think it's better to send the output to the Decoder, which could handle the output just like HandleInput does.